backend/src/services/pdf_generator.py

```
@@ -398,14 +516,30 @@ def _create_daily_schedule(self, schedule: List[Dict]) -> List:
             name = self._safe_text(restaurant.get("name", "Restaurant"))
             story.append(Paragraph(f"<b>{name}</b>", self.styles['SubHeader']))
+
+            # Photo if available
+            photo = restaurant.get("photo") or (restaurant.get("photos") or [None])[0]
+            if photo:
+                try:
+                    # Use synchronous helper to avoid complicating flow
+                    img_path = self._image_cache.get(photo)
+                    if not img_path:
+                        with tempfile.NamedTemporaryFile(delete=False, suffix='.jpg') as tmp:
+                            urllib.request.urlretrieve(photo, tmp.name)
```

Ruff (0.12.2)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)